src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -184,7 +204,50 @@ class EventsRelation(config: RelationConfig)(@transient val sqlContext: SQLConte
     (postUpdate, postNewItems).parMapN((_, _) => ())
   }
 
-  override val fieldsWithPushdownFilter = Seq("type", "subtype", "assetIds", "startTime", "source")
+  override val fieldsWithPushdownFilter: Seq[String] =
+    Seq("type", "subtype", "assetIds", "startTime", "source")
+
+  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
+    val b = urlWithBody(filters, byIdUrl)
+    val u = urlsWithFilters(filters, listUrl)
+
+    if (b.body.nonEmpty) {
+      CdpEventsByIdsRdd(
+        sqlContext.sparkContext,
+        (e: EventItem) => {
+          if (config.collectMetrics) {
+            itemsRead.inc()
+          }
+          toRow(e, requiredColumns)
+        },
+        config,
+        partitionUrlBodies(b)
+      )
+    } else {
+      super.buildScan(requiredColumns, filters)
```

Comment:
We don't call `super.buildScan` anywhere else, why do we need it here? What does it do?

Reply:
The urlWithBody gets the filters on `"id", "externalId"`. If these are not present, then we would like to call super.buildscan which does pushdown on the other filters `"type", "subtype", "assetIds", "startTime", "source"`.

Reply:
I see, this would be easier to follow if we checked for `id` filters separately and said `if (idFilters.nonEmpty) { CdpEventsByIdsRdd(...)` and so on.